library/core/src/pin.rs

```diff
@@ -781,6 +781,34 @@ impl<'a, T: ?Sized> Pin<&'a mut T> {
     }
 }
 
+impl<T: ?Sized> Pin<&'static T> {
+    /// Get a pinned reference from a static reference.
+    ///
+    /// This is safe, because the `'static` lifetime guarantees the data will
+    /// never be moved.
```

Comment:
I think this could be better worded. It's not the `'static` lifetime itself that guarantees the data will never be moved:

```
let static_mut: &'static mut u8 = Box::leak(Box::new(1));
mem::replace(static_mut, 2);
```

...but rather that `Pin::static_ref` borrows the data for the `'static` lifetime, which never ends.

Maybe: 

```suggestion
    /// This is safe, because `T` is borrowed for the `'static` lifetime, which never ends.
```

Reply:
Thanks!

This is part of #77726. Applied the change there.

Reply:
Looks good!

FWIW for a moment I thought `replace` would make it unsafe - the new wording would have corrected my own misunderstanding!